### PR TITLE
Update CI with new compilers & configurations

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: .gitlab/lassen-jobs.yml [hunk 1]
```diff
@@ -5,153 +5,153 @@
...
 
-pgi_19_10:
+xl_16_1_1_10:
   variables:
-    SPEC: "+fortran%pgi@19.10"
+    SPEC: "+fortran %xl@16.1.1.10"
   extends: .build_and_test_on_lassen
 
-pgi_19_10_gcc_8_3_1:
+pgi_19_10_l:
```
Comment (kab163) [inline .gitlab/lassen-jobs.yml:38]: What is the "l" for? Or is that a typo?

Comment (davidbeckingsale) [reply .gitlab/lassen-jobs.yml:38]: Good question. I'll have to double check. It shouldn't impact that actual specs being tested, just the job name that shows up in GitLab.

**Comment by adrienbernede** [comment]
@davidbeckingsale I am tracking existing diffs with develop before I revamp the CI config.
What is the status of this ?

**Comment by davidbeckingsale** [comment]
This PR got stale, and will be redone. I can do that after you have restructured things.

**Pull request closed without merging**